### PR TITLE
PerMessageDeflate#cleanup - properly cleanup on close

### PR DESCRIPTION
**Pull request by yosiat**
Solves #1617

### CODE REVIEW

File: lib/permessage-deflate.js [hunk 1]
```diff
@@ -438,6 +446,11 @@ class PerMessageDeflate {
 
       if (fin) data = data.slice(0, data.length - 4);
 
+      // Set callback to null so we won't call it twice if socket is closed
```
Comment (lpinca) [inline lib/permessage-deflate.js]:
```suggestion
      //
      // Ensure that the callback will not be called again in
      // `PerMessageDeflate#cleanup()`.
      //
```

File: lib/permessage-deflate.js [hunk 2]
```diff
@@ -438,6 +446,11 @@ class PerMessageDeflate {
...
+      // (see PerMessageDeflate#cleanup)
```
Comment (lpinca) [inline lib/permessage-deflate.js]:
```suggestion
```

File: lib/permessage-deflate.js [hunk 3]
```diff
@@ -438,6 +446,11 @@ class PerMessageDeflate {
...
+      // ideally, we would set it to `null` before calling, but we might set it to null
```
Comment (lpinca) [inline lib/permessage-deflate.js]:
```suggestion
```

**Comment by yosiat** [comment]
@lpinca applied your suggestions, can we merge it and release a version?
I want to use it in production (probably I'll need to fork engine.io)

**Comment by lpinca** [comment]
Yes will do in a bit.

**Comment by yosiat** [comment]
@lpinca thanks for all the help!!

**Comment by lpinca** [comment]
Thank you.

**Pull request merged**